examples/cpp/grpc/pingpong/pingpong.cpp

```diff
@@ -0,0 +1,276 @@
+    auto threadPool = std::make_shared<bond::ext::gRPC::thread_pool>();
+
+    DoublePingServiceImpl double_ping_service;
+    PingPongServiceImpl ping_pong_service;
+
+    bond::ext::gRPC::server_builder builder;
+    builder.SetThreadPool(threadPool);
+    const std::string server_address("127.0.0.1:50051");
+    builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+    builder.RegisterService(&double_ping_service)
+        .RegisterService(&ping_pong_service);
+    std::unique_ptr<bond::ext::gRPC::server> server(builder.BuildAndStart());
+
+    DoublePing::Client doublePing(
+        grpc::CreateChannel(server_address, grpc::InsecureChannelCredentials()),
+        ioManager,
+        threadPool);
+
+    PingPong<PingRequest>::Client pingPong(
+        grpc::CreateChannel(server_address, grpc::InsecureChannelCredentials()),
+        ioManager,
+        threadPool);
+
+
+    const std::string user("pong");
+
+    countdown_event ping_event(5);
+
+    bool isCorrectResponse;
+    bool pingNoPayloadIsCorrectResponse;
+    bool pingVoidIsCorrectResponse;
+    bool pingShouldThrowIsCorrectResponse;
+    bool pingGenericIsCorrectResponse;
+
+    ClientContext context;
+    ClientContext pingNoPayloadContext;
+    ClientContext pingNoResponseContext;
+    ClientContext pingVoidContext;
+    ClientContext pingShouldThrowContext;
+    ClientContext pingGenericContext;
+
+    PingRequest request;
+    request.name = user;
+    bond::bonded<PingRequest> req(request);
+
+    {
+        auto f_print = [&ping_event, &isCorrectResponse](bond::bonded<PingReply> response, Status status)
+            {
+                printAndSet(&ping_event, &isCorrectResponse, response, status);
+            };
+
+        doublePing.AsyncPing(&context, req, f_print);
+    }
+
+    {
+        auto f_print = [&ping_event, &pingNoPayloadIsCorrectResponse](bond::bonded<PingReply> response, Status status)
+            {
+                printAndSet(&ping_event, &pingNoPayloadIsCorrectResponse, response, status);
+            };
+
+        doublePing.AsyncPingNoPayload(&pingNoPayloadContext, f_print);
+    }
+
+    {
+        doublePing.AsyncPingNoResponse(&pingNoResponseContext, req);
+    }
+
+    {
+        auto f_print = [&ping_event, &pingVoidIsCorrectResponse](bond::bonded<bond::Void> response, Status status)
+            {
+                if(status.ok())
+                {
+                    pingVoidIsCorrectResponse = true;
+                    ping_event.set();
+                }
+                else
+                {
+                    pingVoidIsCorrectResponse = false;
+                }
+            };
+
+        doublePing.AsyncPingVoid(&pingVoidContext, f_print);
+    }
+
+    {
+        auto f_print = [&ping_event, &pingShouldThrowIsCorrectResponse](bond::bonded<PingReply> response, Status status)
+            {
+                pingShouldThrowIsCorrectResponse = false;
+
+                if(!status.ok())
```

Comment:
We can skip the `.ok()` check and just compare the `error_code()`. (`.ok()` is a helper for `error_code() == StatusCode::OK`)